handlers/vault.go

```diff
@@ -0,0 +1,122 @@
+package handlers
+
+import (
+	"net/http"
+
+	"github.com/gaia-pipeline/gaia/security"
+
+	"github.com/labstack/echo"
+)
+
+type secret struct {
+	Key   string `json:"key"`
+	Value string `json:"value"`
+}
+
+type updateSecret struct {
+	Key   string `json:"key"`
+	Value string `json:"newvalue"`
+}
+
+// UpdateSecret updates a secret using the vault.
+func UpdateSecret(c echo.Context) error {
```

Comment:
I don't like this duplication. I'm going to pull it into SetSecret which updates or Creates.